### ISSUE TITLE
Enable wifi and captive portal in case no ethernet connection could be established

### ISSUE THREAD

**Issue by ChristianWaechter** [esphome/feature-requests#1357]
**Describe the problem you have/What new integration you would like**
I'm using a ESP32 with integrated ethernet connectivity (LILYGO TTGO T-Internet-POE as esp-wrover-kit). This works fine with the ethernet component of type LAN8720. When trying to enable WIFI and ethernet, ESPhome will show the error "Component ethernet cannot be used together with component wifi.". It might be nice to have the possibility to enable both at the same time for connection redundancy, but I guess this will not work due to too few recources or shared pins. But one of the following options would be still nice:
- First try to establish a connection via ethernet. If this fails, start the wifi and try to connect wirelessly
- Have the ESPs wifi access point with the captive portal starting, in case no ethernet connection or wifi could be established

**Please describe your use case for this integration and alternatives you've tried:**

When working on the sensor while it is not connected via ethernet you now have no way of accessing it which can make flashing complicated. Imagine you want to flash a ESP with ethernet which already has a old or unknown firmware on it. Due to the maybe unknown IP settings of the old firmware, you cannot access it via ethernet but need to flash via USB. When the ESP is build into a sealed housing in a dirty environment (like in my case), you cannot just demount the ESP32 and connect USB to it. But you could connect to its wifi access point, if it would start in case of the missing ethernet connection.

**Additional context**

**Comment by vitucciog** [comment]
I do agree... Also we should have the possibility to use captive portal I case of faulty Ethernet connection...

**Comment by FloMeyer** [comment]
+1

**Comment by SurfingSpider** [comment]
+1

**Comment by BBart19** [comment]
+1

**Comment by lupusx** [comment]
+1

**Comment by aznel** [comment]
Half of this has landed. Coexistence is in: [esphome/esphome#14255](https://github.com/esphome/esphome/pull/14255) lifted CONFLICTS_WITH = ["wifi"] and added network: priority: (merged 2026-07-23, released in 2026.8.0), so ethernet: + wifi: with a priority list naming both now compiles on ESP32.

The AP half does not appear to be. Reading 2026.8.2 source, the wifi component raises its fallback AP on wifi state alone — ~90 s after the radio last associated, regardless of Ethernet. A board with a healthy cable and a rotated SSID will broadcast an AP indefinitely beside a working link. We work around it with ap_timeout: 0s and raising the AP ourselves only once both interfaces have been down.

So: "wifi as a configured fallback" — done. "AP only when no interface is up" — still open, and now cleanly expressible on top of the priority list.
